forwards_paper.tex

```diff
@@ -373,52 +447,106 @@ \subsection*{Recording the pedigree in forwards time}
-rather than randomly generated.
-An example is shown in Figure \ref{fig:method_diagram}.
-
-In this scheme,
-at any point in the simulation genealogical history is recorded in a tree sequence.
-This has two additional advantages.
-First, simplification can be run periodically through the simulation,
-taking the set of samples to be the entire currently alive population.
-This is important as it keeps memory usage from growing linearly (and quickly) with time.
-Second, the simulation can be \emph{begun} with a tree sequence produced by some other method --
-for instance, by a coalescent simulation with \msprime.
-This allows for incorporation of deep-time history beyond the reach of individual-based simulations.
-Since geographic structure from times longer ago than the mixing time of migration across the range
-has limited effect on modern genealogies \citep{wilkins_separation}
-(other than possibly changing effective population size \citet{durretspatial}),
-this may not negatively affect realism.
-
-\plr{Something about putting mutations down on the tree sequence.}
+of the current day population. We can use the process of simplification here to
+reduce this down to the minimal set of nodes and edges required to represent
+the information that we are interested in. We can also use simplification if  we have some very
+large tree sequence representing a large dataset and we wish to extract the
+information relevant to a subset of the samples.
+
+Roughly, simplification works by tracing ancestry from the samples backwards
+through the recorded history, adding node and edge records to the output
+only when coalescent events are reached. This works exactly as in \msprime,
+allowing substantial re-use of algorithms; the main difference being that
+parental choice, mutations, and recombination locations are determined by the
+input tree sequence rather than randomly generated. An example is shown in
+Figure \ref{fig:method_diagram}.
+
+The approach that we take is based on Hudson's algorithm for simulating
+the coalescent with
+recombination~\citep{hudson1983properties,kelleher2016efficient}. The main
+state of the algorithm is a set of ancestral segments; each segment $(\ell, r,
+u)$ represents a tract of ancestral material in which the node $u$ is
+associated with the genomic interval $[\ell, r)$. In the same manner
+as Hudson's algorithm, we represent the set of extant lineages as a
+set of individuals, where each ancestral individual is composed of a
+linked list of these segments. The node $u$ in each ancestral sequence
+is the node ID associated with this segment in the output tree sequence.
+We maintain a map $A$ such that $A_j$ is segment chain for node $j$ in
+the input tree sequence.
+
+\jk{I'm not sure this is the right level of detail to tackle this description
+at. What's the goal of this algorithm description? Are we trying to
+establish some properties of the algorithm or just give the reader a high-level
+flavour of how it works??}
```

Comment:
At least a high-level flavour - this is a good start.  I think it'd be nice to outline the steps in at least a bit more detail, to make it more concrete.  I suppose it also depends where we submit this, how much detail is optimal.

Reply:
This is a good question, which we should probably resolve soonish. I don't have any strong views, I'm almost certainly happy with whatever you think is best.